### DISCUSSION TITLE
Filament change sensor

### DISCUSSION THREAD

**Discussion by jpr3d** [Jyers/Marlin#351] [Ideas]
Is it possible to get a filiament change sensor working with this firmware?
The firmware looks great!!

Johan

**Comment by joeclifford** [comment]
It already does, it has been working since release 1.2.0.

https://github.com/Jyers/Marlin/releases

**Reply by jpr3d** [reply]
That was a quick response ;-)
I installed the default firmware on your gthub. Is that the correct one?

**Reply by joeclifford** [reply]
Yep, the current version, 1.2.2, has an option in Control / Advanced to enable/disable the filament sensor.

There is also an option to change the runout distance.

**Reply by jpr3d** [reply]
I checked that option but it doesn't seem to respond.
I will test it again.

**Reply by jpr3d** [reply]
It works now. I had to change low into high.
Thank you for the firmware.

**Reply by marcomania75** [reply]
can you please explain? I have the same issue and I do not understand your solution :o)

**Reply by marcomania75** [reply]
I might figured it out, you meant the variable at:

#define FIL_RUNOUT_STATE     LOW

but for me it was another solution, I had to activate:

 #define FILAMENT_MOTION_SENSOR

**Comment by jpr3d** [comment]
I was just looking it up. Correct. I changed that to get the correct setting from the sensor. If you use FILAMENT_MOTION you probably use a wheel sensor from BigTreeTech?

Johan

________________________________

**Reply by marcomania75** [reply]
correct i have the smart filament sensor from BTT and two days ago I did not know that there is a difference in marlin in how the sensors are configured :-)

**Comment by jpr3d** [comment]
Do you have it working?

Groeten


Johan


________________________________

**Reply by marcomania75** [reply]
Yes, activating

#define FILAMENT_MOTION_SENSOR

was enough, I did not have to change the FIL-RUNOUT_STATE

I only tested it for running out of filament, not for filament getting stuck

**Comment by jpr3d** [comment]
Thank you. I will try that.

Groeten


Johan


________________________________
